include/libboostasio.h

```diff
@@ -161,7 +160,13 @@ class LibBoostAsioHandler : public virtual TcpHandler
                                             shared_from_this(),
 #endif
                                             connection,
-                                            fd)));
+                                            fd);
+                _socket.async_read_some(
+                    boost::asio::null_buffers(),
+                    [f = func, strand = _strand](const boost::system::error_code &ec, const std::size_t bytes_transferred) {
+                        strand_sock_dispatch_func(f, strand, ec, bytes_transferred);
+                    }
+                );
```

Comment:
We could minimise code duplication by adding a member function like the following:

```
std::function<void(boost::system::error_code,std::size_t)> get_io_handler(const bool bRead,
																		  TcpConnection *const connection,
																		  const int fd)
{
	auto func = boost::bind(bRead ? &Watcher::read_handler : &Watcher::write_handler,
							this,
							boost::arg<1>(),
							boost::arg<2>(),
// C++17 has 'weak_from_this()' support.
#if __cplusplus >= 201701L
							weak_from_this(),
#else
							shared_from_this(),
#endif
							connection,
							fd);
	return [f = func, strand = _strand](const boost::system::error_code &ec, const std::size_t bytes_transferred)
	{
		strand_sock_dispatch_func(f, strand, ec, bytes_transferred);
	};
}
```

...the you could just call:

`
_socket.async_write_some(boost::asio::null_buffers(), get_io_handler(true, connection, fd));
`

You should then be able to do something very similar for the timer

Note: Not sure if this compiles.

Reply:
I like the suggestion - will see if it can be integrated.